<h1 style="color:blue; font-size:30px;">📊 Retail Analytics – 🎮 Gaming Sales</h1>

<h2 style="color:black; font-size:22px;">Phase 1: Data Loading and Pre-Processing</h2>

Aim : The aim of this project is to analyze global video game sales data, understand market trends, and generate insights into the performance of different consoles, genres, and publishers.

Problem Definition:The video game industry includes multiple consoles, genres, publishers, and developers competing across different regions (North America, Japan, PAL regions, and Others). However, sales performance varies significantly across platforms and regions.

<h2 style="font-size:20px; color:black; font-weight:bold; font-style:italic;">
📚 Step 1: Preparing the Environment:Library Import
</h2>

In [1]:
import pandas as pd
import numpy as np

print("Libraries imported successfully!")

Libraries imported successfully!


<h2 style="font-size:20px; color:black; font-weight:bold; font-style:italic;">
💾 Step 2: Loading the Dataset
</h2>

In [3]:
df = pd.read_csv("gaming sales.csv")
df.head()

,img,title,console,genre,publisher,developer,critic_score,total_sales,na_sales,jp_sales,pal_sales,other_sales,release_date,last_update
0,/games/boxart/full_6510540AmericaFrontccc.jpg,Grand Theft Auto V,PS3,Action,Rockstar Games,Rockstar North,9.4,20.32,6.37,0.99,9.85,3.12,9/17/2013,NaN
1,/games/boxart/full_5563178AmericaFrontccc.jpg,Grand Theft Auto V,PS4,Action,Rockstar Games,Rockstar North,9.7,19.39,6.06,0.60,9.71,3.02,11/18/2014,1/3/2018
2,/games/boxart/827563ccc.jpg,Grand Theft Auto: Vice City,PS2,Action,Rockstar Games,Rockstar North,9.6,16.15,8.41,0.47,5.49,1.78,10/28/2002,NaN
3,/games/boxart/full_9218923AmericaFrontccc.jpg,Grand Theft Auto V,X360,Action,Rockstar Games,Rockstar North,NaN,15.86,9.06,0.06,5.33,1.42,9/17/2013,NaN
4,/games/boxart/full_4990510AmericaFrontccc.jpg,Call of Duty: Black Ops 3,PS4,Shooter,Activision,Treyarch,8.1,15.09,6.18,0.41,6.05,2.44,11/6/2015,1/14/2018


<h2 style="font-size:20px; color:black; font-weight:bold; font-style:italic;">
📊 Step 3: Dataset Overview (Shape & Columns)
</h2>

In [5]:
rows, columns = df.shape

print("Number of Rows:", rows)
print("Number of Columns:", columns)


Number of Rows: 64016
Number of Columns: 14


<h2 style="font-size:20px; color:black; font-weight:bold; font-style:italic;">
🔎 Step 4: Inspecting Dataset Information
</h2>

In [6]:
df.columns


Index(['img', 'title', 'console', 'genre', 'publisher', 'developer',
       'critic_score', 'total_sales', 'na_sales', 'jp_sales', 'pal_sales',
       'other_sales', 'release_date', 'last_update'],
      dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64016 entries, 0 to 64015
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   img           64016 non-null  object 
 1   title         64016 non-null  object 
 2   console       64016 non-null  object 
 3   genre         64016 non-null  object 
 4   publisher     64016 non-null  object 
 5   developer     63999 non-null  object 
 6   critic_score  6678 non-null   float64
 7   total_sales   18922 non-null  float64
 8   na_sales      12637 non-null  float64
 9   jp_sales      6726 non-null   float64
 10  pal_sales     12824 non-null  float64
 11  other_sales   15128 non-null  float64
 12  release_date  56965 non-null  object 
 13  last_update   17879 non-null  object 
dtypes: float64(6), object(8)
memory usage: 6.8+ MB


Insights
* Several columns contain missing values:• developer → 17 missing entries
* critic_score → only ~6,678 non-null out of 64,016 (very sparse)
* total_sales and regional sales columns (na_sales, jp_sales, pal_sales, other_sales) → large portions missing
* release_date and last_update → many missing values as well

In [8]:
df.describe()


,critic_score,total_sales,na_sales,jp_sales,pal_sales,other_sales
count,6678.000000,18922.000000,12637.000000,6726.000000,12824.000000,15128.000000
mean,7.220440,0.349113,0.264740,0.102281,0.149472,0.043041
std,1.457066,0.807462,0.494787,0.168811,0.392653,0.126643
min,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,6.400000,0.030000,0.050000,0.020000,0.010000,0.000000
50%,7.500000,0.120000,0.120000,0.040000,0.040000,0.010000
75%,8.300000,0.340000,0.280000,0.120000,0.140000,0.030000
max,10.000000,20.320000,9.760000,2.130000,9.850000,3.120000


Overview
* The dataset is highly skewed: a small number of games achieve massive sales, while the majority sell modestly.
* Critic scores cluster around 7–8, suggesting most games receive average-to-good reviews.
* This highlights that blockbuster titles (like GTA V) drive overall sales figures, while most games remain niche.

In [9]:
df.isnull().sum()

img                 0
title               0
console             0
genre               0
publisher           0
developer          17
critic_score    57338
total_sales     45094
na_sales        51379
jp_sales        57290
pal_sales       51192
other_sales     48888
release_date     7051
last_update     46137
dtype: int64

In [10]:
df.duplicated().sum()

np.int64(0)

In [11]:
df['release_date'] = pd.to_datetime(df['release_date'], errors='coerce')
df['release_year'] = df['release_date'].dt.year

# Phase 1 Key Insights

- The dataset contains 64,000 rows and 14 columns.
- It includes categorical, numerical, and datetime features.
- Missing values are present in critic_score column.
- No duplicate records found.
- Date column successfully converted to datetime format.
- Dataset is ready for preprocessing in Phase 2.


In [11]:
import pandas as pd
import pandas as pd

# Load the original CSV into a DataFrame
df = pd.read_csv('gaming sales.csv')

# Check the first few rows to make sure it loaded
df.head()

,img,title,console,genre,publisher,developer,critic_score,total_sales,na_sales,jp_sales,pal_sales,other_sales,release_date,last_update
0,/games/boxart/full_6510540AmericaFrontccc.jpg,Grand Theft Auto V,PS3,Action,Rockstar Games,Rockstar North,9.4,20.32,6.37,0.99,9.85,3.12,9/17/2013,NaN
1,/games/boxart/full_5563178AmericaFrontccc.jpg,Grand Theft Auto V,PS4,Action,Rockstar Games,Rockstar North,9.7,19.39,6.06,0.60,9.71,3.02,11/18/2014,1/3/2018
2,/games/boxart/827563ccc.jpg,Grand Theft Auto: Vice City,PS2,Action,Rockstar Games,Rockstar North,9.6,16.15,8.41,0.47,5.49,1.78,10/28/2002,NaN
3,/games/boxart/full_9218923AmericaFrontccc.jpg,Grand Theft Auto V,X360,Action,Rockstar Games,Rockstar North,NaN,15.86,9.06,0.06,5.33,1.42,9/17/2013,NaN
4,/games/boxart/full_4990510AmericaFrontccc.jpg,Call of Duty: Black Ops 3,PS4,Shooter,Activision,Treyarch,8.1,15.09,6.18,0.41,6.05,2.44,11/6/2015,1/14/2018


In [14]:
# Replace 'your_file.csv' with your actual CSV file name
df = pd.read_csv('gaming sales.csv')

# Check the first few rows
df.head()

,img,title,console,genre,publisher,developer,critic_score,total_sales,na_sales,jp_sales,pal_sales,other_sales,release_date,last_update
0,/games/boxart/full_6510540AmericaFrontccc.jpg,Grand Theft Auto V,PS3,Action,Rockstar Games,Rockstar North,9.4,20.32,6.37,0.99,9.85,3.12,9/17/2013,NaN
1,/games/boxart/full_5563178AmericaFrontccc.jpg,Grand Theft Auto V,PS4,Action,Rockstar Games,Rockstar North,9.7,19.39,6.06,0.60,9.71,3.02,11/18/2014,1/3/2018
2,/games/boxart/827563ccc.jpg,Grand Theft Auto: Vice City,PS2,Action,Rockstar Games,Rockstar North,9.6,16.15,8.41,0.47,5.49,1.78,10/28/2002,NaN
3,/games/boxart/full_9218923AmericaFrontccc.jpg,Grand Theft Auto V,X360,Action,Rockstar Games,Rockstar North,NaN,15.86,9.06,0.06,5.33,1.42,9/17/2013,NaN
4,/games/boxart/full_4990510AmericaFrontccc.jpg,Call of Duty: Black Ops 3,PS4,Shooter,Activision,Treyarch,8.1,15.09,6.18,0.41,6.05,2.44,11/6/2015,1/14/2018


In [17]:
df = pd.read_csv(r'C:\Users\hp\Desktop\gaming sales.csv')

In [20]:
cleaned_df = df.dropna().drop_duplicates()

# Check first few rows of cleaned data
cleaned_df.head()

,img,title,console,genre,publisher,developer,critic_score,total_sales,na_sales,jp_sales,pal_sales,other_sales,release_date,last_update
1,/games/boxart/full_5563178AmericaFrontccc.jpg,Grand Theft Auto V,PS4,Action,Rockstar Games,Rockstar North,9.7,19.39,6.06,0.60,9.71,3.02,11/18/2014,1/3/2018
4,/games/boxart/full_4990510AmericaFrontccc.jpg,Call of Duty: Black Ops 3,PS4,Shooter,Activision,Treyarch,8.1,15.09,6.18,0.41,6.05,2.44,11/6/2015,1/14/2018
7,/games/boxart/full_4653215AmericaFrontccc.jpg,Red Dead Redemption 2,PS4,Action-Adventure,Rockstar Games,Rockstar Games,9.8,13.94,5.26,0.21,6.21,2.26,10/26/2018,11/2/2018
8,/games/boxart/full_1977964AmericaFrontccc.jpg,Call of Duty: Black Ops II,X360,Shooter,Activision,Treyarch,8.4,13.86,8.27,0.07,4.32,1.20,11/13/2012,4/7/2018
9,/games/boxart/full_4649679AmericaFrontccc.png,Call of Duty: Black Ops II,PS3,Shooter,Activision,Treyarch,8.0,13.80,4.99,0.65,5.88,2.28,11/13/2012,4/7/2018


In [21]:
import os
os.listdir()  # should show 'cleaned_data.csv' if saved in notebook folder

['.ipynb_checkpoints',
 '01_Data_Loading_and_Overview-2.ipynb',
 '01_Data_Loading_and_Overview.ipynb',
 'Data Pre Processing.ipynb',
 'desktop.ini',
 'Exploratory Data Analysis.ipynb',
 'gaming sales.csv',
 'GoTo.lnk',
 'hilton.sabacloud.com_content_prodencryption_aOeuVMgiUt6lB2nz758ohg_1764798576_0216MDEzNnd1RGlVM25XZXo5VVRUQjBOQUFVSWJxcGlQVnlxSmhiQUxMUjNlT3JtQ0FLa3ZSbHVCbUpMOXhyTXU0cGNHTFp1SmRUb1VHN1c0VDc5ODBVZSt0eXhLSE9PakZUWngrenhrRFlaRnZkQ1ZZcUdYT2krSEowRXNvRndsb.pdf',
 'Python Assignment 2.ipynb',
 'Smart Browser.lnk',
 'SmartPOS_v5.LNK',
 'sms_exp - Shortcut (2).lnk',
 'SNOONU STOCK.xlsx',
 'STORE PRE- OPENING CHECKLIST.docx',
 'WhatsApp Image 2026-02-19 at 1.04.49 AM.jpeg']

In [24]:
new_df = pd.read_csv('gaming sales.csv')
new_df.head()

,img,title,console,genre,publisher,developer,critic_score,total_sales,na_sales,jp_sales,pal_sales,other_sales,release_date,last_update
0,/games/boxart/full_6510540AmericaFrontccc.jpg,Grand Theft Auto V,PS3,Action,Rockstar Games,Rockstar North,9.4,20.32,6.37,0.99,9.85,3.12,9/17/2013,NaN
1,/games/boxart/full_5563178AmericaFrontccc.jpg,Grand Theft Auto V,PS4,Action,Rockstar Games,Rockstar North,9.7,19.39,6.06,0.60,9.71,3.02,11/18/2014,1/3/2018
2,/games/boxart/827563ccc.jpg,Grand Theft Auto: Vice City,PS2,Action,Rockstar Games,Rockstar North,9.6,16.15,8.41,0.47,5.49,1.78,10/28/2002,NaN
3,/games/boxart/full_9218923AmericaFrontccc.jpg,Grand Theft Auto V,X360,Action,Rockstar Games,Rockstar North,NaN,15.86,9.06,0.06,5.33,1.42,9/17/2013,NaN
4,/games/boxart/full_4990510AmericaFrontccc.jpg,Call of Duty: Black Ops 3,PS4,Shooter,Activision,Treyarch,8.1,15.09,6.18,0.41,6.05,2.44,11/6/2015,1/14/2018


In [4]:
import pandas as pd

# Load the dataset
df = pd.read_csv("gaming sales.csv")

# Clean the dataset (drop missing values + duplicates)
cleaned_df = df.dropna().drop_duplicates()

# Save cleaned dataset to Excel
cleaned_df.to_excel("cleaned_gaming_sales.xlsx", index=False)

print("✅ Cleaned dataset saved as cleaned_gaming_sales.xlsx")

✅ Cleaned dataset saved as cleaned_gaming_sales.xlsx


In [5]:
import os
file_path = os.path.abspath("cleaned_gaming_sales.xlsx")
print("📂 Excel file location:", file_path)

📂 Excel file location: C:\Users\hp\Desktop\cleaned_gaming_sales.xlsx
